app/models/classification.rb

```diff
@@ -395,7 +398,7 @@ def self.import_from_hash(classification, parent = nil)
 
     stats = {"categories" => 0, "entries" => 0}
 
-    if classification["parent_id"] == 0 # category
+    if parent.nil?
```

Comment:
So the hash contains (or might contain?) a `"parent_id"` key, and we get a `parent` arg as well?
Checking `parent` looks fine, but are the two always in sync?  Below (L411) we `create(classification)` which would use `classification["parent_id"]`, ignoring `parent`.

Looking at db/fixtures/classifications.yml, categories come with `:parent_id: 0`, and entries without `:parent_id`.
OK, I think I see.  The recursive call goes into the other branch, which sets `"parent_id" => parent.id`.

=> Assuming the imported yaml is always rooted at categories, LGTM.
  Consider also forcing "parent_id" => parent.id in the categories branch too.

P.S. db/fixtures/classifications.yml is not really relevant.  That file is loaded by `.seed` which uses completely separate code :man_facepalming:

Reply:
This import_from_yaml->import_from_hash code path is only used by `tools/import_tags.rb`.  I don't think we have tests or examples what input looks like, although it's pretty obvious it should be symmetric to `Classification.export_to_yaml`.  That one looks like the fixture — always rooted at categories, omitting `"parent_id"` from entries => satisfies assumptions here :+1:

There is also a 3rd completely separate importing code in `lib/task_helpers/imports/tags.rb`.
And a 4th  completely separate https://github.com/rhtconsulting/cfme-rhconsulting-scripts/blob/master/rhconsulting_tags.rake 
:confused: :man_facepalming: :laughing: 
oh well, consolidating is totally not the goal here, I'm just amused.

Reply:
However you also need to update lib/task_helpers/exports/tags.rb, which has `:parent_id => 0`.  
lib/task_helpers/exports/tags.rb looks unaffected to me?